svc/svc.go

```diff
@@ -0,0 +1,246 @@
+// Package svc provides some tooling to make building services with remind101/pkg
+// easier.
+//
+// Recommend Usage:
+//
+//	func main() {
+//		env := svc.InitAll()
+//		defer env.Close()
+//
+//		r := httpx.NewRouter()
+//		// ... add routes
+//
+//		h := svc.NewStandardHandler(svc.HandlerOpts{
+//			Router:   r,
+//			Reporter: env.Reporter,
+//	})
+//
+// 	svc.RunServer(h, "80", 5*time.Second)
+// }
+package svc
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	ddtrace "github.com/DataDog/dd-trace-go/opentracing"
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/pkg/errors"
+	"github.com/remind101/pkg/httpx"
+	"github.com/remind101/pkg/httpx/middleware"
+	"github.com/remind101/pkg/logger"
+	"github.com/remind101/pkg/metrics"
+	"github.com/remind101/pkg/reporter"
+	"github.com/remind101/pkg/reporter/rollbar"
+)
+
+type HandlerOpts struct {
+	Router            *httpx.Router
+	Reporter          reporter.Reporter
+	ForwardingHeaders []string
+	BasicAuth         string
+}
+
+// NewStandardHandler returns an http.Handler with a standard middleware stack.
+// The last middleware added is the first middleware to handle the request.
+// Order is pretty important as some middleware depends on others having run
+// already.
+func NewStandardHandler(opts HandlerOpts) http.Handler {
```

Comment:
Not sure if this is overreaching, but this could also handle the `/health` route since that is standard across all servers.